lib/socket.c

```diff
@@ -3063,6 +3064,14 @@ uc_socket_listen(uc_vm_t *vm, size_t nargs)
 	if (fd == -1)
 		err_return(errno, "socket()");
 
+	if (ucv_is_truish(reuseaddr)) {
+		int yes = 1;
+
+		ret = setsockopt(fd, SOL_SOCKET, SO_REUSEADDR, &yes, sizeof(yes));
```

Comment:
Please use a compound literal here and put an empty line between the statement and the subsequent `if (...)`:

```c
ret = setsockopt(fd, SOL_SOCKET, SO_REUSEADDR, &(int){ 1 }, sizeof(int));

if (ret == -1)
    err_return(errno, "setsockopt()");
```

Also - does it make sense to also set `SO_REUSEPORT` if available?

Reply:
Not as a new option, just wondered if it might make sense to cover both with the same option, but thinking again about it, it is too uncommon, users who need this can setup the socket manually using lower level methods along with the appropriate setopt() calls.